test/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js

```diff
@@ -0,0 +1,79 @@
+const TESTHUBENDPOINT_URL = `http://${document.location.host}/testhub`;
+
+describe('hubConnection', () => {
+    eachTransport(transportName => {
+        it(`over ${transportName} can invoke server method and receive result`, done => {
+            const message = "Hi";
+            let hubConnection = new signalR.HubConnection(TESTHUBENDPOINT_URL, 'formatType=json&format=text');
+
+            hubConnection.start(transportName)
+                .then(() => {
+                    hubConnection.invoke('Microsoft.AspNetCore.SignalR.Test.Server.TestHub.Echo', message)
+                    .then(result => {
+                        expect(result).toBe(message);
+                    })
+                    .catch(() => {
+                        expect(false).toBe(true);
+                    })
+                    .then(() => {
+                        hubConnection.stop();
+                        done();
+                    })
+                })
+                .catch(() => {
+                    expect(false).toBe(true);
+                    done();
+                })
+        });
+
+        it(`over ${transportName} rethrows an exception from the server`, done => {
+            const errorMessage = "An error occurred.";
+            let hubConnection = new signalR.HubConnection(TESTHUBENDPOINT_URL, 'formatType=json&format=text');
+
+            hubConnection.start(transportName)
+                .then(() => {
+                    hubConnection.invoke('Microsoft.AspNetCore.SignalR.Test.Server.TestHub.ThrowException', errorMessage)
+                        .then(() => {
+                            // exception expected but none thrown
+                            expect(true).toBe(false);
```

Comment:
jasmine does not have `fail` so for the purpose of testing both are equivalent as the mean to fail the test. I think I could just create a fail() function and replace all these.